File: PatronInterprete/Introduccion_Interprete/src/Simbolo/Tipo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Simbolo;

/**
 *
 * @author JohnAlds
 */
public class Tipo {
    private tipoInstruccion tipo;
    private tipoDato dato;

    public Tipo(tipoInstruccion tipo) {
        this.tipo = tipo;
    }
    
    public Tipo(tipoDato dato) {
        this.dato = dato;
    }

    public tipoInstruccion getTipo() {
        return tipo;
    }

    public void setTipo(tipoInstruccion tipo) {
        this.tipo = tipo;
    }
    
    
    public tipoDato getTipoDato() {
        return dato;
    }

    public void setTipoDato(tipoDato dato) {
        this.dato = dato;
    }
    
    
}

File: Jison/Simbolo/Tipo.ts
import { tipoInstruccion } from "./tipoInstruccion";
import { tipoDato } from "./tipoDato";

export class Tipo {

    private _tipo?: tipoInstruccion
    private _dato?: tipoDato
    private _dimensiones: number

    constructor(valor: tipoInstruccion | tipoDato, esDato: boolean = true, dimensiones: number = 0) {
        if (esDato) {
            this._dato = valor as tipoDato
        } else {
            this._tipo = valor as tipoInstruccion
        }
        this._dimensiones = dimensiones
    }

    get tipo(): tipoInstruccion | undefined {
        return this._tipo
    }

    set tipo(valor: tipoInstruccion | undefined) {
        this._tipo = valor
    }

    get tipoDato(): tipoDato | undefined {
        return this._dato
    }

    set tipoDato(valor: tipoDato | undefined) {
        this._dato = valor
    }

    get dimensiones(): number {
        return this._dimensiones
    }

    set dimensiones(valor: number) {
        this._dimensiones = valor
    }
}